Cargo.toml

```diff
@@ -10,34 +10,16 @@ cxx = "1.0.78"
 # without disabling default features, cargo will complain `libz-sys` feature is required but vector doesn't have it
 # for release, use "vector = { git = "https://github.com/yanhuangdata/vector.git", branch = "feature/vector-lib" }"
 # for local repo test, use "vector = { path = "{path_to_local_vector_repo}" }"
-vector = { git = "https://github.com/yanhuangdata/vector.git", branch = "feature/vector_lib_19", features = [
-    "api",
-    "api-client",
-    "disk-buffer",
-    "enrichment-tables",
-    "rdkafka-plain",
-    "sinks",
-    "sources",
-    "sources-dnstap",
-    "transforms",
-    "unix",
-    "vrl-cli",
-    "openssl",
-    "vendor-sasl",
-], default-features = false }
+vector = { git = "https://github.com/yanhuangdata/vector.git", branch = "feature/vector-upgrade", default-features = true }
```

Comment:
We could probably remove some of the sources/sinks we are unlikely to use (e.g. datadog), but I would like to keep transforms unchanged, otherwise, it may cause some trouble difficult to diagnose, (e.g. some of the transforms cannot be used even following docs correctly).

One of the issue is `rdkafka?/gssapi-vendored` feature is weirdly named, and I don't figure out what the meaning is for this naming convention in cargo